packages/teams-js/src/public/tasks.ts

```diff
@@ -36,8 +41,11 @@ export namespace tasks {
         (sdkResponse: dialog.ISdkResponse) => submitHandler(sdkResponse.err, sdkResponse.result)
       : undefined;
     if (taskInfo.card !== undefined || taskInfo.url === undefined) {
-      ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.meetingStage);
```

Comment:
Not a comment on your change, but I am wondering why we did not throw an error here when both card and url and provided, but instead waited until it reaches to the host SDK?

Reply:
AFAIK we wanted the host to resolve it. Don't know if this the exact reason.

Reply:
If something is going to fail, we want to fail it sooner. I learned it from @TrevorJoelHarris. Leaving this open in case Trevor wants to comment on it.

Reply:
Failing fast is always good. In this specific case, this function was originally written by Teams and in order to maintain backwards compatibility with their existing function we don't want to make it fail either at a different *time* than it used to or in a different *way* than it used to. Apps that calls tasks.startTask should see an error come from teams if they specify both card and url, but since we don't know what that error is we'll continue to let Teams handle it. We should *definitely* fail fast in our new dialog capability though (or even better, do what Shreyas is doing in this change and make failure not possible by making different function signatures that don't allow for invalid parameter combinations)

Reply:
It is always backwards compatibility :/
Thank you for commenting! :)